lib/diagnostic.js

```diff
@@ -36,7 +57,10 @@ const newArray = function newArray(length, defaultValue) {
 // The function you write should return
 // the object passed in as an argument or the object created
 const addMethod = function addMethod(propertyName, method, obj) {
-
+  if (typeof obj [method] === 'undefined') {
+  obj[method] = propertyName;
+  return true;
```

Comment:
This function does not return the modified object. Additionally, you've confused `propertyName` (the key) and `method` (the value). Finally, you haven't accounted for what happens if `obj` is undefined (i.e. no value is passed in as a third argument).